### ISSUE TITLE
Sleep Timer Alternative - audio book to night music transition

### ISSUE THREAD

**Issue by c33s** [flackbash/AudioAnchor#58]
currently the sleep timer is a normal sleep timer which simply stops playing after a given time.

my idea/wish would be a little bit different: if you want to fall asleep with an audio book but want to sleep with white noise/relaxing noises like water sounds or wood sounds it would be awesome to have a special timer that can be set and after the timer finished the sound just not stop but jumps to a different folder (audio book).

lets say we have the following folder structure, having 3 audio books (2 "real" audio books  and 1 wood noise "audio book".
a config section where the "music timer" folder can be set would be handsome. setting the "music timer" to 30 minutes in the while listening to "php done right" and previously set the "music timer" folder to "wood sound 1h" the php done right would be played for 30min and then wood sound 1h would be played for the rest of the night (as long as nobody press stop or the battery is empty).
``` 
audio books:
    php done right
        chapter1.mp3
        chapter2.mp3
        chapter3.mp3
        ...
    perry rhodan
        chapter1.mp3
        chapter2.mp3
        chapter3.mp3
        ...
    wood sound 1h
        wood.mp3      
```

i would be more than happy to have such a feature.

**Comment by flackbash** [comment]
I agree that something like this would be a cool feature. It will be quite some work to implement, so I won't promise anything but I'll definitely keep it in mind. 

In the meantime, do you have a concrete idea what this config section could look like?
Just a list of all available albums or books from which you can choose the "sleep book" together with a timer how long it should play and an option to have it play indefinitely in a loop?
Your suggestion is to select an entire folder as "sleep book", right. I would find it more consistent if one would choose a single track. Then, also to be consistent, the tracks that would be played afterwards should depend on the autoplay settings,

**Comment by SzymonPopowicz** [comment]
Hey, on google playstore there is an app called smart audiobook. It has my most favourite feature where I cannot find in any other audiobook readers - a sleep timer where in the last minute the volume is lineary going down, rather then just stop playing after a certain amount of time. Then if you shake your phone and the timer resets and the volume gets back to normal.

With this feature you can go to sleep while listening to an audiobook without having to look at the screen at night to reset the timer. Then if you fall asleep you can just jump back the time of your timer and youre good.

**Comment by flackbash** [comment]
AudioAnchor has the same feature ;) In the settings menu under *Sleep timer* you can adjust the fade out time (10s by default) to 60s and enable *Shake to reset sleep timer*. There is also a slider to adjust the shake sensitivity. Enable *Vibrate on reset by shake* to get haptic feedback if the sleep timer was reset by shake.

**Comment by c33s** [comment]
> In the meantime, do you have a concrete idea what this config section could look like?

for me it will be most handy to simply select a whole directory as sleep book, maybe as dropdown of all albums in the settings or maybe by longpressing an album select it and "mark as sleepbook".

> Just a list of all available albums or books from which you can choose the "sleep book" 

yes maybe is a list/dropdown the best option

> together with a timer how long it should play and an option to have it play indefinitely in a loop?

for me the whole point is that a sleep book folder (with multiple tracks) it plays indefinitly. if i would need a timer i would use the normal sleep timer.

it's about i go to settings, mark a folder/album as "sleep book" or "night sound" or "after timer indefinitly play until i press stop", enable the feature in the settings, then go to a completly different album like my audio book which simply let me listen to a good night story, which is played as long as i have choosen with the regular sleep timer, lets say 30mins, after that time the audio book stops and the music switches to the "night sound" which contains something like https://www.youtube.com/watch?v=o8GrqUSdzi0&list=RDQMZYoFBLFpYI8&start_radio=1 and https://www.youtube.com/watch?v=1ZYbU82GVz4 (just many tracks like that) which keeps playing until wake up again.
good would also be to choose a specific volume for the night sound, so i can have for example higher volume for the audio book and lower for the night sound.
```
[30min audio book ][endless multi tracks play until stopped manually]
```

> Your suggestion is to select an entire folder as "sleep book", right. I would find it more consistent if one would choose a single track.
this would be impractible, as i have a folder with multiple short meditative and some 3h long tracks, if i can only choose one i would be required to build a single track from my 40 night sound tracks. if i am able to choose an album it doesn't matter if it only contains one or multiple tracks.

> Then, also to be consistent, the tracks that would be played afterwards should depend on the autoplay settings,

no, no other tracks should play, only indefinitly the sleep sound album. 

imagine you peacefully fall asleep with a nice fairytale, you fly away in you dreams carried by "[Relaxing Sleep Music: Deep Sleeping Music, Relaxing Music, Stress Relief, Meditation Music][1]"  just to be beaten out of your dreams by [Slipknot - Nero Forte][2]  ;)

it's all about having the nice sound played all the time, so if you wake up you instantly hear the relaxing music again and float back into your dreams.

[1]: https://www.youtube.com/watch?v=1ZYbU82GVz4
[2]: https://www.youtube.com/watch?v=JGNqvH9ykfA

**Comment by ange007** [comment]
Is it possible to implement a fade in sound throughout the duration of the sleep timer?

Sorry to interrupt, but this is also good for sleeping. I'm just looking at the player. A gradual change in volume is the simplest and most effective option (if this player allows it to be done normally, since on others the volume changes not very smoothly, and with large steps).

**Comment by c33s** [comment]
@ange007 sounds like a new issue/ticket for me.
